app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt

```diff
@@ -76,4 +96,14 @@ class FormView @JvmOverloads constructor(
         )
         layoutManager.scrollToPositionWithOffset(myFirstPositionIndex, offset)
     }
+
+    private fun checkLastItem(): Boolean {
+        val layoutManager =
+            recyclerView.layoutManager as LinearLayoutManager
+        val lastVisiblePosition = layoutManager.findLastVisibleItemPosition()
+        return lastVisiblePosition != -1 && (
+            lastVisiblePosition == adapter.itemCount - 1 ||
+                adapter.getItemViewType(lastVisiblePosition) == R.layout.form_section
+            )
```

Comment:
reviewing how to improve this

Reply:
I managed to get a similar behaviour using this: 

private fun checkLastItem(v: View): Boolean {
       
        if (!v.canScrollVertically(DOWN)){
            return true
        }
        return false
}

could I miss anything related to adapter.getItemViewType(lastVisiblePosition) == R.layout.form_section ? Not sure why was needed to be honest. I thought this adapter.itemCount - 1  could be enough

Reply:
No, i think this would be enough 👍

Reply:
yes, maybe with `adapter.itemCount - 1` should work. Have you tested it?